### ISSUE TITLE
Parallelize the flat Karatsuba implementation

### ISSUE THREAD

**Issue by emmorais** [Inversed-Tech/eyelid#35]
Parallelize flat Karatsuba: 

- [ ] Each layer of the flat Karatsuba computation has independent chunks that can be calculated completely in parallel. The results must be ready for next layer computation. 
- [ ] When we get too close to the top of the calculation tree, parallelization will be less efficient, then we need to fine tune a parameter that will dictate when to stop using parallelization.

**Comment by teor2345** [comment]
I've marked this as `wontfix`, because:
- The recursive implementation is 4-5x faster: https://github.com/Inversed-Tech/eyelid/pull/32#issuecomment-2054782381
- Parallelising computations that take a few milliseconds usually doesn't provide much performance gain, because of thread overhead

Instead, we could parallelise multiple separate iris comparisons or polynomial multiplications.

**Comment by teor2345** [comment]
We’re more likely to parallelise the overall comparison, rather than individual operations. It’s more efficient to do that.
